### PR TITLE
add OpenAPI spec for the action server

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fixes https://github.com/RasaHQ/rasa_core_sdk/issues/39
- add an OpenAPI spec for the action server endpoint

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: docs/_static/spec/action_server.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+                type: object
+                properties:
+                  action_name:
+                    type: string
+                  error:
+                    type: string
+        500:
+          description: >-
+            The action server encountered an exception while running the action.
+components:
```
Comment (tmbo) [inline docs/_static/spec/action_server.yml]: I think at least some of these are duplicated between the two specifications - how about creating a common file we can import from: seems to be possible to ref components in different files: https://swagger.io/docs/specification/using-ref/
and 
https://dev.to/mikeralphson/defining-reusable-components-with-the-openapi-specification-4077

**Comment by tmbo** [comment]
Is this ready for review?

**Comment by wochinge** [comment]
No, this is not yet ready for review. I have to work on other things first but had to share that early.

**Comment by JustinaPetr** [comment]
@wochinge the docs look good. A little cosmetic thing is that an event name goes a little bit off-limits here:

2019-01-30_09h57_47


Is it something we could easily fix?

**Comment by wochinge** [comment]
@JustinaPetr fixed it by using the event name
bildschirmfoto 2019-01-31 um 10 37 54

File: docs/_static/spec/action_server.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+openapi: "3.0.2"
+info:
+  title: "Rasa Core SDK - Action Server Endpoint"
+  version: "1.0.0"
```
[the remark this answers is not in the document]
Comment (wochinge) [reply docs/_static/spec/action_server.yml]: mhm, `0.12.1` would make more sense 👍

[the remark this answers is not in the document]
Comment (tmbo) [reply docs/_static/spec/action_server.yml]: I'd actually rather like to remove it: the whole documentation is versioned, and as long as we do not have any automation to update this version number, it is going to go stale

[the remark this answers is not in the document]
Comment (wochinge) [reply docs/_static/spec/action_server.yml]: ok. then I remove it from `server.yml` as wel

[the remark this answers is not in the document]
Comment (tmbo) [reply docs/_static/spec/action_server.yml]: :+1:

[the remark this answers is not in the document]
Comment (wochinge) [reply docs/_static/spec/action_server.yml]: Damn, this is a required parameter. Setting it to `""` seems a bit evil, doesn't it?

**Pull request merged**